src/prefect/_internal/compatibility/experimental.py

```diff
@@ -179,6 +182,57 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def experimental_field(
+    name: str,
+    *,
+    group: str,
+    help: str = "",
+    stacklevel: int = 2,
+    opt_in: bool = False,
+    when: Optional[Callable[[Any], bool]] = None,
+):
+    """
+    Mark a field in a Pydantic model as experimental.
+
+    Raises warning only if the field is specified during init.
+
+    Example:
+
+        ```python
+
+        @experimental_parameter("y", group="example", when=lambda y: y is not None)
+        def foo(x, y = None):
+            return x + 1 + (y or 0)
+        ```
+    """
+
+    when = when or (lambda _: True)
+
+    @experimental(
+        group=group,
+        feature=f"The field {name!r}",
+        help=help,
+        opt_in=opt_in,
+        stacklevel=stacklevel + 2,
+    )
+    def experimental_check():
+        pass
+
+    def decorator(model_cls: Type[M]) -> Type[M]:
+        cls_init = model_cls.__init__
+
+        def __init__(__pydantic_self__, **data: Any) -> None:
```

Comment:
Did you confirm that this decorator logic preserves docstrings / typing / etc as we would expect?  If not, would be useful to use `functools.wraps`

Reply:
Good call. I added a wraps decorator.